<launch>
	<node name="controller" pkg="pid" type="controller" ns="motorFR">
            <rosparam file="$(find robot_hardware)/param/pid.yaml" command="load" />
			<remap from="setpoint" to="desired_vel" />
            <remap from="control_effort" to="pwm" />
            <remap from="state" to="encoderVelocity" />
            <remap from="pid_enable" to="/encoder_enable" />
    </node>

	<node name="controller" pkg="pid" type="controller" ns="motorFL">
            <rosparam file="$(find robot_hardware)/param/pid.yaml" command="load" />
			<remap from="setpoint" to="desired_vel" />
            <remap from="control_effort" to="pwm" />
            <remap from="state" to="encoderVelocity" />
            <remap from="pid_enable" to="/encoder_enable" />
    </node>
    
    <node name="controller" pkg="pid" type="controller" ns="motorBR">
            <rosparam file="$(find robot_hardware)/param/pid.yaml" command="load" />
			<remap from="setpoint" to="desired_vel" />
            <remap from="control_effort" to="pwm" />
            <remap from="state" to="encoderVelocity" />
            <remap from="pid_enable" to="/encoder_enable" />
    </node>
    
    <node name="controller" pkg="pid" type="controller" ns="motorBL">
            <rosparam file="$(find robot_hardware)/param/pid.yaml" command="load" />
			<remap from="setpoint" to="desired_vel" />
            <remap from="control_effort" to="pwm" />
            <remap from="state" to="encoderVelocity" />
            <remap from="pid_enable" to="/encoder_enable" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="motorSensorFR">
            <rosparam file="$(find robot_hardware)/param/pidSensor.yaml" command="load" />
			<remap from="setpoint" to="/desired_pose" />
            <remap from="control_effort" to="/motorFR/pwm" />
            <remap from="state" to="error" />
            <remap from="pid_enable" to="/pid_enable" />
    </node>

	<node name="controller" pkg="pid" type="controller" ns="motorSensorFL">
            <rosparam file="$(find robot_hardware)/param/pidSensor.yaml" command="load" />
            <remap from="setpoint" to="/desired_pose" />
            <remap from="control_effort" to="/motorFL/pwm" />
            <remap from="state" to="error" />
            <remap from="pid_enable" to="/pid_enable" />
    </node>
    
    <node name="controller" pkg="pid" type="controller" ns="motorSensorBR">
            <rosparam file="$(find robot_hardware)/param/pidSensor.yaml" command="load" />
			<remap from="setpoint" to="/desired_pose" />
            <remap from="control_effort" to="/motorBR/pwm" />
            <remap from="state" to="error" />
            <remap from="pid_enable" to="/pid_enable" />
    </node>
    
    <node name="controller" pkg="pid" type="controller" ns="motorSensorBL">
            <rosparam file="$(find robot_hardware)/param/pidSensor.yaml" command="load" />
			<remap from="setpoint" to="/desired_pose" />
            <remap from="control_effort" to="/motorBL/pwm" />
            <remap from="state" to="error" />
            <remap from="pid_enable" to="/pid_enable" />
    </node>

</launch>   
